docs/settings.rst
================
List of settings
================

.. module:: zinnia.settings

Zinnia has a lot of parameters to configure the application accordingly to
your needs. Knowing this list of settings can save you a lot of time.

Here's a full list of all available settings, and their default values.

All settings described here can be found in :file:`zinnia/settings.py`.

.. contents::
    :local:
    :depth: 1

.. _settings-entry:

Entry
=====

.. setting:: ZINNIA_ENTRY_TEMPLATES

ZINNIA_ENTRY_TEMPLATES
----------------------
**Default value:** ``()`` (Empty tuple)

List of tuple for extending the list of templates availables for
rendering the entry. By using this setting, you can change the look and
feel of an entry directly in the admin interface. Example: ::

  ZINNIA_ENTRY_TEMPLATES = (('zinnia/entry_detail_alternate.html',
                             gettext('Alternative template')),)

.. setting:: ZINNIA_ENTRY_BASE_MODEL

ZINNIA_ENTRY_BASE_MODEL
-----------------------
**Default value:** ``''`` (Empty string)

String defining the base model path for the Entry model. See
:doc:`extending_entry_model` for more informations.

.. setting:: ZINNIA_UPLOAD_TO

ZINNIA_UPLOAD_TO
----------------
**Default value:** ``'uploads/zinnia'``

String setting that tells Zinnia where to upload entries' images.

.. versionchanged:: 0.10
Previously the default value was ``'uploads'``

.. _settings-edition:

Edition
=======

.. setting:: ZINNIA_MARKUP_LANGUAGE

ZINNIA_MARKUP_LANGUAGE
----------------------
**Default value:** ``'html'``

String determining the markup language used for writing the entries.
You can use one of these values: ::

    ['html', 'markdown', 'restructuredtext', 'textile']

The value of this variable will alter the value of :setting:`ZINNIA_WYSIWYG`
if you don't set it.

.. setting:: ZINNIA_MARKDOWN_EXTENSIONS

ZINNIA_MARKDOWN_EXTENSIONS
--------------------------
**Default value:** ``''`` (Empty string)

Extensions names to be used for rendering the entries in MarkDown. Example:
::

  ZINNIA_MARKDOWN_EXTENSIONS = 'extension1_name,extension2_name...'

.. setting:: ZINNIA_WYSIWYG

ZINNIA_WYSIWYG
--------------
**Default value:** ::

    WYSIWYG_MARKUP_MAPPING = {
        'textile': 'markitup',
        'markdown': 'markitup',
        'restructuredtext': 'markitup',
        'html': 'tinymce' in settings.INSTALLED_APPS and \
                    'tinymce' or 'wymeditor'}

    WYSIWYG = getattr(settings, 'ZINNIA_WYSIWYG',
                      WYSIWYG_MARKUP_MAPPING.get(ZINNIA_MARKUP_LANGUAGE))

Determining the WYSIWYG editor used for editing an entry.
So if MarkDown, Textile or reStructuredText are used, the value will be
``'markitup'``, but if you use HTML, TinyMCE will be used if
:ref:`django-tinymce is installed<zinnia-tinymce>`, else WYMEditor will be
used.

This setting can also be used for disabling the WYSIWYG
functionnality. Example: ::

  ZINNIA_WYSIWYG = None

.. _settings-views:

Views
=====

.. setting:: ZINNIA_PAGINATION

ZINNIA_PAGINATION
-----------------

**Default value:** ``10``

Integer used to paginate the entries. So by default you will have 10
entries displayed per page on the Weblog.

.. setting:: ZINNIA_ALLOW_EMPTY

ZINNIA_ALLOW_EMPTY
------------------
**Default value:** ``True``

Used for archives views, raise a 404 error if no entries are present at
a specified date.

.. setting:: ZINNIA_ALLOW_FUTURE

ZINNIA_ALLOW_FUTURE
-------------------
**Default value:** ``True``

Used for allowing archives views in the future.

.. _settings-feeds:

Feeds
=====

.. setting:: ZINNIA_FEEDS_FORMAT

ZINNIA_FEEDS_FORMAT
-------------------
**Default value:** ``'rss'``

String determining the format of the syndication feeds. You can use
``'atom'`` if your prefer Atom feeds.

.. setting:: ZINNIA_FEEDS_MAX_ITEMS

ZINNIA_FEEDS_MAX_ITEMS
----------------------
**Default value:** ``15``

Integer used to define the maximum items provided in the syndication feeds.
So by default you will have 15 entries displayed on the feeds.

.. _settings-urls:

URLs
====

.. setting:: ZINNIA_URL_SHORTENER_BACKEND

ZINNIA_URL_SHORTENER_BACKEND
----------------------------
**Default value:** ``'zinnia.url_shortener.backends.default'``

String representing the module path to the URL shortener backend.

.. setting:: ZINNIA_PROTOCOL

ZINNIA_PROTOCOL
---------------
**Default value:** ``'http'``

String representing the protocol of the site. If your Web site uses HTTPS,
set this setting to ``https``.

.. _settings-comments:

Comment moderation
==================

.. setting:: ZINNIA_AUTO_MODERATE_COMMENTS

ZINNIA_AUTO_MODERATE_COMMENTS
-----------------------------
**Default value:** ``False``

Determine if a new comment should be allowed to show up
immediately or should be marked non-public and await approval.

.. setting:: ZINNIA_AUTO_CLOSE_COMMENTS_AFTER

ZINNIA_AUTO_CLOSE_COMMENTS_AFTER
--------------------------------
**Default value:** ``None``

Determine the number of days where comments are open. If you set this
setting to ``10`` the comments will be closed automaticaly 10 days after
the publication date of your entries.

.. setting:: ZINNIA_MAIL_COMMENT_REPLY

ZINNIA_MAIL_COMMENT_REPLY
-------------------------
**Default value:** ``False``

Boolean used for sending an email to comment's authors
when a new comment is posted.

.. setting:: ZINNIA_MAIL_COMMENT_AUTHORS

ZINNIA_MAIL_COMMENT_AUTHORS
---------------------------
**Default value:** ``True``

Boolean used for sending an email to entry authors
when a new comment is posted.

.. setting:: ZINNIA_MAIL_COMMENT_NOTIFICATION_RECIPIENTS

ZINNIA_MAIL_COMMENT_NOTIFICATION_RECIPIENTS
-------------------------------------------
**Default value:** ::

    [manager_tuple[1] for manager_tuple in settings.MANAGERS]

List of emails used for sending a notification when a
new public comment has been posted.

.. setting:: ZINNIA_SPAM_CHECKER_BACKENDS

ZINNIA_SPAM_CHECKER_BACKENDS
----------------------------
**Default value:** ``()`` (Empty tuple)

List of strings representing the module path to a spam checker backend.
See :doc:`spam_checker` for more informations about this setting.

.. _settings-pinging:

Pinging
=======

.. setting:: ZINNIA_PING_DIRECTORIES

ZINNIA_PING_DIRECTORIES
-----------------------
**Default value:** ``('http://django-blog-zinnia.com/xmlrpc/',)``

List of the directories you want to ping.

.. setting:: ZINNIA_PING_EXTERNAL_URLS

ZINNIA_PING_EXTERNAL_URLS
-------------------------
**Default value:** ``True``

Boolean setting for telling if you want to ping external URLs when saving
an entry.

.. setting:: ZINNIA_SAVE_PING_DIRECTORIES

ZINNIA_SAVE_PING_DIRECTORIES
----------------------------
**Default value:** ``bool(ZINNIA_PING_DIRECTORIES)``

Boolean setting for telling if you want to ping directories when saving
an entry.

.. setting:: ZINNIA_PINGBACK_CONTENT_LENGTH

ZINNIA_PINGBACK_CONTENT_LENGTH
------------------------------
**Default value:** ``300``

Size of the excerpt generated on pingback.

.. _settings-similarity:

Similarity
==========

.. setting:: ZINNIA_F_MIN

ZINNIA_F_MIN
------------
**Default value:** ``0.1``

Float setting of the minimal word frequency for similar entries.

.. setting:: ZINNIA_F_MAX

ZINNIA_F_MAX
------------
**Default value:** ``1.0``

Float setting of the minimal word frequency for similar entries.

.. _settings-misc:

Miscellaneous
=============

.. setting:: ZINNIA_COPYRIGHT

ZINNIA_COPYRIGHT
----------------
**Default value:** ``'Zinnia'``

String used for copyrighting your entries, used in the syndication feeds
and in the opensearch document.

.. setting:: ZINNIA_STOP_WORDS

ZINNIA_STOP_WORDS
-----------------
**Default value:** See :file:`zinnia/settings.py`

List of common words excluded from the advanced search engine
to optimize the search querying and the results.

.. setting:: ZINNIA_USE_TWITTER

ZINNIA_USE_TWITTER
------------------
**Default value:** ``True if python-twitter is in the PYTHONPATH``

Boolean telling if Zinnia can use Twitter.

.. _settings-cms:

CMS
===

All the settings related to the CMS can be found in :file:`zinnia/plugins/settings.py`.

.. setting:: ZINNIA_APP_MENUS

ZINNIA_APP_MENUS
----------------
**Default value:** ::

  ('zinnia.plugins.menu.EntryMenu',
   'zinnia.plugins.menu.CategoryMenu',
   'zinnia.plugins.menu.TagMenu',
   'zinnia.plugins.menu.AuthorMenu')

List of strings representing the path to the Menu class provided for the
Zinnia AppHook.

.. setting:: ZINNIA_HIDE_ENTRY_MENU

ZINNIA_HIDE_ENTRY_MENU
----------------------
**Default value:** ``True``

Boolean used for displaying or not the entries in the EntryMenu object.

.. setting:: ZINNIA_PLUGINS_TEMPLATES

ZINNIA_PLUGINS_TEMPLATES
------------------------
**Default value:** ``()`` (Empty tuple)

List of tuple for extending the CMS's plugins rendering templates.


zinnia/settings.py
"""Settings of Zinnia"""
from django.conf import settings

PING_DIRECTORIES = getattr(settings, 'ZINNIA_PING_DIRECTORIES',
                           ('http://django-blog-zinnia.com/xmlrpc/',))
SAVE_PING_DIRECTORIES = getattr(settings, 'ZINNIA_SAVE_PING_DIRECTORIES',
                                bool(PING_DIRECTORIES))
SAVE_PING_EXTERNAL_URLS = getattr(settings, 'ZINNIA_PING_EXTERNAL_URLS', True)

COPYRIGHT = getattr(settings, 'ZINNIA_COPYRIGHT', 'Zinnia')

PAGINATION = getattr(settings, 'ZINNIA_PAGINATION', 10)
ALLOW_EMPTY = getattr(settings, 'ZINNIA_ALLOW_EMPTY', True)
ALLOW_FUTURE = getattr(settings, 'ZINNIA_ALLOW_FUTURE', True)

ENTRY_TEMPLATES = getattr(settings, 'ZINNIA_ENTRY_TEMPLATES', [])
ENTRY_BASE_MODEL = getattr(settings, 'ZINNIA_ENTRY_BASE_MODEL', '')

MARKUP_LANGUAGE = getattr(settings, 'ZINNIA_MARKUP_LANGUAGE', 'html')

MARKDOWN_EXTENSIONS = getattr(settings, 'ZINNIA_MARKDOWN_EXTENSIONS', '')

WYSIWYG_MARKUP_MAPPING = {
    'textile': 'markitup',
    'markdown': 'markitup',
    'restructuredtext': 'markitup',
    'html': 'tinymce' in settings.INSTALLED_APPS and 'tinymce' or 'wymeditor'}

WYSIWYG = getattr(settings, 'ZINNIA_WYSIWYG',
                  WYSIWYG_MARKUP_MAPPING.get(MARKUP_LANGUAGE))

AUTO_CLOSE_COMMENTS_AFTER = getattr(
    settings, 'ZINNIA_AUTO_CLOSE_COMMENTS_AFTER', None)

AUTO_MODERATE_COMMENTS = getattr(settings, 'ZINNIA_AUTO_MODERATE_COMMENTS',
                                 False)

MAIL_COMMENT_REPLY = getattr(settings, 'ZINNIA_MAIL_COMMENT_REPLY', False)

MAIL_COMMENT_AUTHORS = getattr(settings, 'ZINNIA_MAIL_COMMENT_AUTHORS', True)

MAIL_COMMENT_NOTIFICATION_RECIPIENTS = getattr(
    settings, 'ZINNIA_MAIL_COMMENT_NOTIFICATION_RECIPIENTS',
    [manager_tuple[1] for manager_tuple in settings.MANAGERS])

UPLOAD_TO = getattr(settings, 'ZINNIA_UPLOAD_TO', 'uploads/zinnia')

PROTOCOL = getattr(settings, 'ZINNIA_PROTOCOL', 'http')

FEEDS_FORMAT = getattr(settings, 'ZINNIA_FEEDS_FORMAT', 'rss')
FEEDS_MAX_ITEMS = getattr(settings, 'ZINNIA_FEEDS_MAX_ITEMS', 15)

PINGBACK_CONTENT_LENGTH = getattr(settings,
                                  'ZINNIA_PINGBACK_CONTENT_LENGTH', 300)

F_MIN = getattr(settings, 'ZINNIA_F_MIN', 0.1)
F_MAX = getattr(settings, 'ZINNIA_F_MAX', 1.0)

SPAM_CHECKER_BACKENDS = getattr(settings, 'ZINNIA_SPAM_CHECKER_BACKENDS',
                                ())

URL_SHORTENER_BACKEND = getattr(settings, 'ZINNIA_URL_SHORTENER_BACKEND',
                                'zinnia.url_shortener.backends.default')

STOP_WORDS = getattr(settings, 'ZINNIA_STOP_WORDS',
                     ('able', 'about', 'across', 'after', 'all', 'almost',
                      'also', 'among', 'and', 'any', 'are', 'because', 'been',
                      'but', 'can', 'cannot', 'could', 'dear', 'did', 'does',
                      'either', 'else', 'ever', 'every', 'for', 'from', 'get',
                      'got', 'had', 'has', 'have', 'her', 'hers', 'him', 'his',
                      'how', 'however', 'into', 'its', 'just', 'least', 'let',
                      'like', 'likely', 'may', 'might', 'most', 'must',
                      'neither', 'nor', 'not', 'off', 'often', 'only', 'other',
                      'our', 'own', 'rather', 'said', 'say', 'says', 'she',
                      'should', 'since', 'some', 'than', 'that', 'the',
                      'their', 'them', 'then', 'there', 'these', 'they',
                      'this', 'tis', 'too', 'twas', 'wants', 'was', 'were',
                      'what', 'when', 'where', 'which', 'while', 'who', 'whom',
                      'why', 'will', 'with', 'would', 'yet', 'you', 'your'))

TWITTER_CONSUMER_KEY = getattr(settings, 'TWITTER_CONSUMER_KEY', '')
TWITTER_CONSUMER_SECRET = getattr(settings, 'TWITTER_CONSUMER_SECRET', '')
TWITTER_ACCESS_KEY = getattr(settings, 'TWITTER_ACCESS_KEY', '')
TWITTER_ACCESS_SECRET = getattr(settings, 'TWITTER_ACCESS_SECRET', '')

USE_TWITTER = getattr(settings, 'ZINNIA_USE_TWITTER',
                      bool(TWITTER_ACCESS_KEY and TWITTER_ACCESS_SECRET and \
                           TWITTER_CONSUMER_KEY and TWITTER_CONSUMER_SECRET))
